rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -56,53 +57,32 @@ uninstall_signal_handlers()
   return SignalHandler::get_global_signal_handler().uninstall();
 }
 
+static
 std::vector<std::string>
-init_and_remove_ros_arguments(
-  int argc,
+_remove_ros_arguments(
   char const * const argv[],
-  const InitOptions & init_options)
+  const rcl_arguments_t * args,
+  rcl_allocator_t alloc)
 {
-  init(argc, argv, init_options);
-  return remove_ros_arguments(argc, argv);
-}
-
-std::vector<std::string>
-remove_ros_arguments(int argc, char const * const argv[])
-{
-  rcl_allocator_t alloc = rcl_get_default_allocator();
-  rcl_arguments_t parsed_args = rcl_get_zero_initialized_arguments();
-
   rcl_ret_t ret;
-
-  ret = rcl_parse_arguments(argc, argv, alloc, &parsed_args);
-  if (RCL_RET_OK != ret) {
-    exceptions::throw_from_rcl_error(ret, "failed to parse arguments");
-  }
-
   int nonros_argc = 0;
   const char ** nonros_argv = NULL;
 
   ret = rcl_remove_ros_arguments(
     argv,
-    &parsed_args,
+    args,
     alloc,
     &nonros_argc,
     &nonros_argv);
 
   if (RCL_RET_OK != ret || nonros_argc < 0) {
     // Not using throw_from_rcl_error, because we may need to append deallocation failures.
-    exceptions::RCLErrorBase base_exc(ret, rcl_get_error_state());
+    exceptions::RCLError exc(ret, rcl_get_error_state(), "");
```

Comment:
why this change?

Reply:
to keep `init_and_remove_ros_arguments` throw the same exception type as the previous original code